x-pack/legacy/plugins/infra/server/new_platform_plugin.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { InternalCoreSetup, PluginInitializerContext } from 'src/core/server';
+import { Server } from 'hapi';
+import { InfraConfig } from './new_platform_config.schema';
+import { Legacy } from '../../../../../kibana';
+import { initInfraServer } from './infra_server';
+import { InfraBackendLibs, InfraDomainLibs } from './lib/infra_types';
+import { FrameworkFieldsAdapter } from './lib/adapters/fields/framework_fields_adapter';
+import { InfraKibanaBackendFrameworkAdapter } from './lib/adapters/framework/kibana_framework_adapter';
+import { InfraKibanaLogEntriesAdapter } from './lib/adapters/log_entries/kibana_log_entries_adapter';
+import { KibanaMetricsAdapter } from './lib/adapters/metrics/kibana_metrics_adapter';
+import { InfraElasticsearchSourceStatusAdapter } from './lib/adapters/source_status';
+import { InfraFieldsDomain } from './lib/domains/fields_domain';
+import { InfraLogEntriesDomain } from './lib/domains/log_entries_domain';
+import { InfraMetricsDomain } from './lib/domains/metrics_domain';
+import { InfraLogAnalysis } from './lib/log_analysis';
+import { InfraSnapshot } from './lib/snapshot';
+import { InfraSourceStatus } from './lib/source_status';
+import { InfraSources } from './lib/sources';
+
+export interface KbnServer extends Server {
+  usage: any;
+}
+
+const DEFAULT_CONFIG: InfraConfig = {
+  enabled: true,
+  query: {
+    partitionSize: 75,
+    partitionFactor: 1.2,
+  },
+};
+
+export class InfraServerPlugin {
+  public config: InfraConfig = DEFAULT_CONFIG;
+  private legacyServer: Legacy.Server;
+  public libs: InfraBackendLibs | undefined;
+
+  constructor(context: PluginInitializerContext, legacyServer: Legacy.Server) {
+    this.legacyServer = legacyServer;
+    const config$ = context.config.create<InfraConfig>();
+    config$.subscribe(configValue => {
+      this.config = {
+        ...DEFAULT_CONFIG,
+        enabled: configValue.enabled,
+        query: {
+          ...DEFAULT_CONFIG.query,
+          ...configValue.query,
```

Comment:
I found that depending on your schema, and whether config options are optional, you might end up with `undefined` values here.